### ISSUE TITLE
Express Checkout - Icons for WC checkout page general list of gateways.

### ISSUE THREAD

**Issue by angelleye** [angelleye/paypal-woocommerce#878]
* https://wordpress.org/support/topic/paypal-png-insecure

In #251 we addressed an issue where the graphic on the checkout is not using https:// when it should.  It looks like those commits were never merged into our core code, though.

Looking at what we do have in our code it seems we did try to address this:

```
    public function get_icon() {
        $image_path = plugins_url('/assets/images/paypal.png', plugin_basename(dirname(__FILE__)));
        if ($this->show_paypal_credit == 'yes') {
            $image_path = plugins_url('/assets/images/paypal-credit-card-logos.png', plugin_basename(dirname(__FILE__)));
        }
        if ($this->checkout_with_pp_button_type == 'customimage') {
            $image_path = $this->pp_button_type_my_custom;
        }
        if (is_ssl()) {
            $image_path = preg_replace("/^http:/i", "https:", $image_path);
        }
        $icon = "<img src=\"$image_path\" alt='" . __('Pay with PayPal', 'paypal-for-woocommerce') . "'/>";
        return apply_filters('woocommerce_paypal_express_icon', $icon, $this->id);
    }
```

I'm a little bit confused by this whole piece of logic.  I'm not sure why we're using paypal.png at all anymore..??  Also, we have paypal-credit.png included in our /assets/images/ but it doesn't seem to be getting used at all in the logic.  We are using paypal-credit-card-logos.png if `show_paypal_credit` is enabled, which is the better graphic, but it doesn't include anything about PayPal Credit.

So we have a few things to look at here.  

* Why would `is_ssl()` be coming up false for this guy when his site is indeed running on https://?  
* Why are we using a graphic within our PayPal Credit specific logic that doesn't actually PayPal Credit image?
* Why do we still have paypal-credit.png included in our assets/images if we aren't using it at all?
* Why are we still using the ugly paypal.png?  I thought we got rid of that a while ago..??

Basically, I think what we need to do here is use the paypal-credit-card-logos.png as our default image instead of paypal.png.  Then, we need to make another version of that which includes a PayPal Credit icon along with the others, and use that when `show_paypal_credit` is enabled.  Finally, we need to double check our logic around http:// vs. https:// to ensure it's always properly output.

**Comment by kcppdevelopers** [comment]
@angelleye we are using paypal.png since last 1.3.0 milestone, please confirm, are you really want to changed express checkout gateway default image to paypal-credit-card-logos.png ?

**Comment by kcppdevelopers** [comment]
@angelleye it seems paypal-credit-card-logos.png image is not suitable for PayPal Express checkout payment method.

**Comment by angelleye** [comment]
@kcppdevelopers Yeah, you're right.  It would make sense if Guest Checkout is enabled to show that you can pay with PayPal or credit cards, but if it's just PayPal then it wouldn't be good to show them all.

I do want to improve that graphic.  It just doesn't look good because it's not centered nicely.  Can we replace it with this one?  You may need to resize it so it fits well.

![paypal-logo](https://user-images.githubusercontent.com/629241/30772084-76e0fcb2-a01a-11e7-92c5-5345448c3edf.png)

**Comment by kcppdevelopers** [comment]
@angelleye right, we need to improve graphic. 

1) if show_paypal_credit is enabled: https://www.paypalobjects.com/webstatic/en_US/i/buttons/cc-badges-ppppcmcvdam.png

2) if Guest Checkout is enabled: https://www.paypalobjects.com/webstatic/en_US/i/buttons/cc-badges-ppmcvdam.png

3) default: https://www.paypalobjects.com/webstatic/en_US/i/buttons/PP_logo_h_100x26.png

**Comment by kcppdevelopers** [comment]
![untitled](https://user-images.githubusercontent.com/13145461/30772615-9073341c-a07c-11e7-8840-04efcf3917e3.png)

**Comment by angelleye** [comment]
@kcppdevelopers How come the PayPal Pro option here doesn't have any graphic/icons next to it?

**Comment by kcppdevelopers** [comment]
![878](https://user-images.githubusercontent.com/13145461/30844566-3fa581d8-a2ad-11e7-9cb7-dc5676cef0f8.png)

it's working well, I was have set wrong path in gateway setting panel.

**Comment by angelleye** [comment]
@kcppdevelopers Please show me again what shows up here for PayPal Express when PayPal Credit is enabled as well?

**Comment by angelleye** [comment]
@kcppdevelopers I checked it out, and when PayPal Credit enabled in settings I'm getting this on the WC checkout page:

![image](https://user-images.githubusercontent.com/629241/30847497-60b44efa-a261-11e7-8d68-224a5801ec7b.png)

So it's still not showing me a graphic that includes PayPal Credit..??

**Comment by angelleye** [comment]
@kcppdevelopers Yeah, seems our logic here still isn't right.  When I disable PayPal Credit I end up with this:

![image](https://user-images.githubusercontent.com/629241/30847579-a74f7a2e-a261-11e7-8865-d3e973cb7b96.png)

However, I do have Guest Checkout enabled, so really the image above is what should show up in this case.

**Comment by angelleye** [comment]
So to reiterate...

1)  If Express Checkout is enabled, but both PayPal Credit and our option to "Allow customers to checkout without a PayPal account using their credit card" are disabled, then we should see the PayPal only graphic.

![image](https://user-images.githubusercontent.com/629241/30847649-e6d0047a-a261-11e7-8165-8d76ca1d58f7.png)

2)  If Express checkout is enabled, PayPal Credit is disabled, but "Allow customers to checkout without a PayPal account using their credit card" is enabled, then we should see the PayPal + CC icons graphic:

![image](https://user-images.githubusercontent.com/629241/30847681-029f3036-a262-11e7-9fd8-9a4e3e115034.png)

3)  If Express is enabled, PayPal Credit is enabled, and "Allow customers to checkout without PayPal..." is enabled, then we should see the PayPal + CC icons graphic, but it needs to also include an icon for PayPal Credit.  I thought I saw you had a graphic like this created already, but I don't see it now..??

**Comment by kcppdevelopers** [comment]
@angelleye I think one more condition we miss here.

4) If Express is enabled, PayPal Credit is enabled, and "Allow customers to checkout without PayPal..." is disabled, then we should see the PayPal +PayPal Credit icon. right?
![screenshot-2017-9-26 checkout dev](https://user-images.githubusercontent.com/13145461/30861680-28611b68-a2e9-11e7-98a6-5ba8fb871357.png)

**Comment by angelleye** [comment]
@kcppdevelopers yes correct
